src/Type/Php/ImplodeFunctionReturnTypeExtension.php

```diff
@@ -35,20 +39,7 @@ public function getTypeFromFunctionCall(
 		if (count($args) === 1) {
 			$argType = $scope->getType($args[0]->value);
 			if ($argType->isArray()->yes()) {
-				$accessoryTypes = [];
-				if ($argType->isIterableAtLeastOnce()->yes() && $argType->getIterableValueType()->isNonEmptyString()->yes()) {
-					$accessoryTypes[] = new AccessoryNonEmptyStringType();
-				}
-				if ($argType->getIterableValueType()->isLiteralString()->yes()) {
-					$accessoryTypes[] = new AccessoryLiteralStringType();
-				}
-
-				if (count($accessoryTypes) > 0) {
-					$accessoryTypes[] = new StringType();
-					return new IntersectionType($accessoryTypes);
-				}
-
-				return new StringType();
```

Comment:
this code was duplicated in the extension. de-duplicated the existing logic with the first commit with https://github.com/phpstan/phpstan-src/pull/991/commits/180b2af22fbefe58e801f313a75acf4ede71b5f4